packages/dds/task-manager/src/test/taskManager.fuzz.spec.ts

```diff
@@ -0,0 +1,422 @@
+                    logCurrentState(state, loggingInfo);
+                    console.log("-".repeat(20));
+                    console.log("Next operation:", JSON.stringify(operation, undefined, 4));
+                }
+                statefulReducer(state, operation);
+                return state;
+            };
+
+    performFuzzActions(
+        generator,
+        {
+            volunteer: statefully(({ clients }, { taskManagerName, taskId }) => {
+                const taskManager = findTaskManager(clients, taskManagerName);
+                assert(taskManager);
+                taskManager.volunteerForTask(taskId).catch((e: Error) => {
+                    // We expect an error to be thrown if we are disconnected while volunteering
+                    const expectedErrors = [
+                        `Disconnected before acquiring task assignment: ${taskId}`,
+                        `Abandoned before acquiring task assignment: ${taskId}`,
+                    ];
+                    if (!expectedErrors.includes(e.message)) {
+                        throw e;
+                    }
+                });
+            }),
+            abandon: statefully(({ clients }, { taskManagerName, taskId }) => {
+                const taskManager = findTaskManager(clients, taskManagerName);
+                assert(taskManager);
+                taskManager.abandon(taskId);
+            }),
+            subscribe: statefully(({ clients }, { taskManagerName, taskId }) => {
+                const taskManager = findTaskManager(clients, taskManagerName);
+                assert(taskManager);
+                taskManager.subscribeToTask(taskId);
+            }),
+            complete: statefully(({ clients }, { taskManagerName, taskId }) => {
+                const taskManager = findTaskManager(clients, taskManagerName);
+                assert(taskManager);
+                taskManager.complete(taskId);
+            }),
+            synchronize: statefully(({ containerRuntimeFactory, clients }) => {
+                containerRuntimeFactory.processAllMessages();
+                assertConsistent(clients);
+            }),
+            changeConnectionState: statefully(({ clients }, { taskManagerName, connected }) => {
+                const containerRuntime = findRuntime(clients, taskManagerName);
+                assert(containerRuntime);
+                containerRuntime.connected = connected;
+            }),
+        },
+        initialState,
+        saveInfo,
+    );
+}
+
+const directory = path.join(__dirname, "../../src/test/results");
+
+function getPath(seed: number): string {
+    return path.join(directory, `${seed}.json`);
+}
+
+const describeFuzz = createFuzzDescribe({ defaultTestCount: defaultOptions.testCount });
+
+describeFuzz("TaskManager fuzz testing", ({ testCount }) => {
+    before(() => {
+        if (!existsSync(directory)) {
+            mkdirSync(directory);
```

Comment:
```suggestion
            mkdirSync(directory, { recursive: true });
```

should give more resilience to future changes at little cost.